pkg/sentry/platform/ring0/pagetables/pagetables.go

```diff
@@ -24,12 +24,23 @@ import (
 	"gvisor.dev/gvisor/pkg/usermem"
 )
 
+const halfSize = 1 + ^uintptr(0)>>1
```

Comment:
This is somewhat mysterious. What do you think about NewWithUpper just taking an upperStart parameter and saving that in the PageTables struct? Then you could just do:

NewWithUpper(NewAllocator(), ring0.KernelStart, m.upperPageTables)

Then it sues ring0.KernelStart directly and is much more clear.